### DISCUSSION TITLE
Should we keep the Margin/Padding CSS utility classes after implementing the styles dynamically.

### DISCUSSION THREAD

**Discussion by owensgit** [Legal-and-General/canopy#509] [General] [answered]
As part of the **[Responsive Padding PR](https://github.com/Legal-and-General/canopy/pull/362)**, all of the CSS rules for our margin and padding directives are being created and added dynamically to a `<style>` tag on the page. 

E.g. when you use `<div lgPadding="sm">`, the following CSS rule is created dynamically:

```
.lg-padding--sm {
  padding: 1rem;
}
```

This means that the directives no longer use the global CSS spacing utility classes, which currently live in **_projects/canopy/src/styles/spacing.scss_** (see screenshot).

![Screenshot 2021-08-06 at 13 08 32](https://user-images.githubusercontent.com/1943532/128508185-5d30a9f7-af2b-4352-bb37-c89573d637fa.png)

**Shall we remove these utility classes?**

Some points to consider

- Would it be a [breaking change](https://github.com/Legal-and-General/canopy/blob/master/docs/BREAKING_CHANGES.md) to remove them? 
- Are these classes useful/used outside of the directives?
- Removing them would save 7kb on the canopy.css file, which is a 12.5% saving

**Comment by Zimann** [comment]
The classes don't seem to be used outside of the `spacing.scss` file, also the file doesn't seem to be imported in other parts of the canopy library. 

Removing them would be nice because then we can just avoid having the 'important' part which would allow us to have a more predictable behavior when it comes to applying the styling. 

However, I have seen that there are references to these variables in other projects which install canopy, and removing them seems to break styling.

**Reply by owensgit** [reply]
Thanks @Zimann, good to know that projects are using these util classes and it wasn't just the spacing directives. That may tip the balance in favour of keeping them.

**Comment by elenagarrone** [comment, marked as answer]
I think we should keep them for a period of time as deprecated. This way the margin and padding directives changes will not be a breaking change. 

I'm pretty sure there are few non angular projects that only use the scss variables. 

One thing we could do is removing the `spacing.scss` file from the global css and add an explicit message on the notes saying that if someone wants to use the classes instead of directives they would have to import the scss file themselves. It would still be a breaking change though.

**Reply by paul-request** [reply]
Yeah I like this idea. I would question whether a note on the release is enough, but then maybe if it breaks other's apps unexpectedly, it is actually down to not-so-good practice in terms of looking at the release notes before upgrading etc.

**Reply by owensgit** [reply]
I like that idea too 👍 

We could add a deprecated note and add them to our list of [upcoming deprecations](https://github.com/Legal-and-General/canopy/issues/315).

Perhaps ahead of the release we'd need to think of how we could communicate upcoming breaking changes to the teams within L&G, I've started a discussion about this: https://github.com/Legal-and-General/canopy/discussions/520
